CMakeLists.txt

```diff
@@ -3,10 +3,10 @@ cmake_minimum_required(VERSION 3.5)
 project(drogon)
 
 message(STATUS "compiler: " ${CMAKE_CXX_COMPILER_ID})
-include(CheckCXXSourceRuns)
-check_cxx_source_runs("int main(){return 0;}" not_cross_compiling)
 
-if (not_cross_compiling)
+# If your cross compile is failing, you should set
+# CMAKE_SYSTEM_NAME in your toolchain file
+if (NOT CMAKE_CROSSCOMPILING)
```

Comment:
According to the CMake documentation, the use of CMAKE_CROSSCOMPILING is not recommended for projects targeting Apple devices, how should we deal with this case?
And why the check_cxx_source_runs doesn't work?

Reply:
Because of underlying `try_run()` behavior, note you can change it by defining `CMAKE_CROSSCOMPILING_EMULATOR` as per [documentation](https://cmake.org/cmake/help/v3.5/command/try_run.html?highlight=try%20run#behavior-when-cross-compiling). 

IMO we should stick with as close to canonical way as possible, so we fit into the ecosystem better.

For apple devices we could try something like
```
if (NOT CMAKE_CROSSCOMPILING)
    include(CheckCXXSourceRuns)
    check_cxx_source_runs("int main(){return 0;}" not_cross_compiling)

    if (not_cross_compiling)
        set(BUILD_PROGRAMS OFF)
    else()
        set(BUILD_PROGRAMS ON)
    endif ()
else ()
...
```
But it's not needed, as by setting `CMAKE_SYSTEM_NAME`, `CMAKE_CROSSCOMPILING` is most definitely turning on, and that is what is the official recommendation. [Source.](https://gitlab.kitware.com/cmake/cmake/-/issues/21885) The docs are just notoriously out of date again, but this is CMake alright xD.